changes/12550-add-validation-for-policy-specs

```diff
@@ -0,0 +1 @@
+- added validation to `fleetctl apply` for duplicate policy names in the yml file and attempting to change the team of an existing policy
```

Comment:
Let's also add another bullet for the API change in `/api/_version_/fleet/spec/policies`.